Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/RestShapeSpringsForceField.inl

```diff
@@ -339,22 +339,24 @@ void RestShapeSpringsForceField<DataTypes>::addForce(const MechanicalParams*  mp
         if (useRestMState)
             ext_index = m_ext_indices[i];
 
+        const auto stiffness = k[static_cast<std::size_t>(i < k.size()) * i];
+
```

Comment:
To me, `k` should be initialized in `init` to be the same size than `m_indices`. There is a check with a warning in `reinit`. In addition to the warning, a resize could be made.
It would avoid the ugly check in the loops.

Reply:
Agreed during the sofadevmeeting on the general design for this kind of code. But outside the scope of the PR.